src/CommentsGridPage.js

```diff
@@ -1,98 +1,36 @@
-import React, { useMemo } from 'react';
-import { groupBy } from 'lodash';
-import moment from 'moment';
+import React, { useEffect, useState } from 'react';
 import { Grid } from '@material-ui/core';
-import { useQuery } from 'react-query';
-import axios from 'axios';
 
 import { Comment } from './components/Comment';
 import { SortBySelector, SortEnum } from './components/SortBySelector';
 
-export const CommentsGridPage = () => {
-  // fetching comments
-  const comments = useQuery(
-    'comments',
-    async () => {
-      const response = await axios.request({
-        method: 'get',
-        url: 'https://www.reddit.com/r/wallstreetbets/comments/.json',
-        params: {
-          sort: 'new',
-          limit: 30, // not 100 for now because of bad performance of fetching user details individually
-        },
-      });
-
-      return response.data?.data?.children?.map((obj) => obj.data);
-    },
-    {
-      keepPreviousData: true, // so the data doesnt disappear
-      refetchInterval: 15 * 1000,
-    }
-  );
-
-  // fetching authors
-  const authors = useQuery(
-    ['authors', { comments: comments.data }],
-    async ({ queryKey: [, { comments }] }) => {
-      // TODO
-      // better to batch fetch the list but not sure how to do that ATM...
-      const authorsDetails = await Promise.all(
-        comments.map(async (comment) => {
-          const response = await axios.get(`https://www.reddit.com/user/${comment.author}/about/.json`);
-          return response.data.data;
-        })
-      );
+import socketIOClient from 'socket.io-client';
+const ENDPOINT = 'http://127.0.0.1:5000/';
 
-      return authorsDetails;
-    },
-    {
-      enabled: !!comments.data,
-    }
-  );
+export const CommentsGridPage = () => {
+  const [comments, setComments] = useState([]);
 
-  // grouping the comments by author criteria
-  // for now grouping by hardcoded account ages
-  const commentGroups = useMemo(() => {
-    if (authors.data && comments.data) {
-      const groups = groupBy(comments.data, (comment) => {
-        const commentAuthorDetails = authors.data.find((author) => comment.author === author.name);
-        const authorAge = moment(commentAuthorDetails.created_utc * 1000);
-        if (authorAge.isBefore(moment().subtract(6, 'years'))) {
-          return 1;
-        } else if (authorAge.isBefore(moment().subtract(3, 'years'))) {
-          return 2;
-        } else if (authorAge.isBefore(moment().subtract(1, 'years'))) {
-          return 3;
-        } else {
-          return 4;
-        }
+  useEffect(() => {
```

Comment:
you should add method to remove socket event listener on unmount. if we add other pages later when a user
`socket.removeAllListeners("new comment");`
 
so 
useEffect(() => {
    //socket connect here
   return () => {
        //remove event listeners here
   }
}, [data, ...comments])

https://stackoverflow.com/questions/9418697/how-to-unsubscribe-from-a-socket-io-subscription